### PR TITLE
Fully Functional App 

### PR DESCRIPTION
**Pull request by thetinayang**
NOTE: for some reason I can only create a pull request onto this branch, **PLEASE DO NOT MERGE IT**

This version of the app is fully functional and only has AutoLayout and UI Bugs left. 
Split View and all the detailViewControllers are now functional. 

BASIC EXPLANATION OF CODE (mainly for Sabrina and Sherin):

- Alongside the SplitView and MasterViewController, there are 3 types of ViewControllers: _AnalyticsViewController, CrashViewController_ and the classic _DetailViewController_. (I adopted Xcode's naming convention)
- _CrashViewController_ and _AnalyticsViewController_ have static UI that is done in Storyboard and presented with segue
- _DetailViewController_ deals with the Build, Test, Distribute and Push pages. They are customized programmatically using the Beacon class. 
- The **Crash page** has two types of crashes, fatalError() and Nil Optional Value
- The **Analytics** page has custom events that will change the text of a label
- All **other pages** will display a corresponding photo, title and description.

### CODE REVIEW

File: demoapp-ios-swift/DetailViewController.swift [hunk 1]
```diff
@@ -10,34 +10,38 @@ import UIKit
 
 class DetailViewController: UIViewController {
 
+    @IBOutlet weak var beaconBlurb: UITextView!
```
Comment (elamalani) [inline demoapp-ios-swift/DetailViewController.swift]: We should not define variables with  "beacon" as prefix in it - anywhere in the code. Can you change it to "serviceBlurb"?

File: demoapp-ios-swift/DetailViewController.swift [hunk 2]
```diff
@@ -10,34 +10,38 @@ import UIKit
...
+        let _  = self.view
         // Update the user interface for the detail item.
-        beaconTitleLabel.text = beacon?.name
-        beaconImageView.image = beacon?.beaconImage()
+        if self.beacon != nil {
+            if let label = beaconTitleLabel {
+                label.text = beacon?.name
+                beaconImageView.image = beacon?.beaconImage()
+                beaconBlurb.text = beacon?.beaconBlurb()
+            }
```
Comment (elamalani) [inline demoapp-ios-swift/DetailViewController.swift]: Replace "beacon" word in all the files.

File: demoapp-ios-swift/DetailViewController.swift [hunk 3]
```diff
@@ -10,34 +10,38 @@ import UIKit
...
     @IBOutlet weak var beaconImageView: UIImageView!
     
     func configureView() {
+        let _  = self.view
         // Update the user interface for the detail item.
-        beaconTitleLabel.text = beacon?.name
-        beaconImageView.image = beacon?.beaconImage()
+        if self.beacon != nil {
+            if let label = beaconTitleLabel {
+                label.text = beacon?.name
```
Comment (MehrdadMzfr) [inline demoapp-ios-swift/DetailViewController.swift]: what not using the `beaconTitleLabel ` directly?

File: demoapp-ios-swift/MasterViewController.swift [hunk 4]
```diff
@@ -11,29 +11,23 @@ import UIKit
 class MasterViewController: UITableViewController {
     
     var detailViewController: DetailViewController? = nil
-    var objects: [Any] = ["Build", "Test", "Distribute", "Crash", "Analytics", "Push"]
-    
-    // construct Data Model array, manually added for now
+    // construct Data Model array
     var beacons: [Beacon] = []
```
Comment (MehrdadMzfr) [inline demoapp-ios-swift/MasterViewController.swift]: services?

File: demoapp-ios-swift/MasterViewController.swift [hunk 5]
```diff
@@ -11,29 +11,23 @@ import UIKit
...
-    
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
         
         if let split = splitViewController {
             let controllers = split.viewControllers
-            detailViewController = (controllers[controllers.count-1] as! UINavigationController).topViewController as? DetailViewController
+            self.detailViewController = (controllers[controllers.count-1] as! UINavigationController).topViewController as? DetailViewController
```
Comment (MehrdadMzfr) [inline demoapp-ios-swift/MasterViewController.swift]: nit: `controllers[controllers.count - 1`
can you check what does the styling tool enforce?

File: demoapp-ios-swift/MasterViewController.swift [hunk 6]
```diff
@@ -71,20 +66,24 @@ class MasterViewController: UITableViewController {
         return self.beacons.count
```
Comment (MehrdadMzfr) [inline demoapp-ios-swift/MasterViewController.swift]: nice!

File: demoapp-ios-swift/MasterViewController.swift [hunk 7]
```diff
@@ -71,20 +66,24 @@ class MasterViewController: UITableViewController {
...
     }
     
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let row = indexPath.row
+        
+        switch row {
```
Comment (MehrdadMzfr) [inline demoapp-ios-swift/MasterViewController.swift]: directly use:
```
switch indexPath.row { 
...
```

**Pull request closed without merging**